web/packages/teleport/src/config.ts

```diff
@@ -686,8 +683,20 @@ const cfg = {
     return searchString ? `${path}?${searchString}` : path;
   },
 
-  getSsoUrl(providerUrl, providerName, redirect) {
-    return cfg.baseUrl + generatePath(providerUrl, { redirect, providerName });
+  getSsoUrl(providerUrl, providerName, redirect, loginHint) {
+      const basePath = cfg.baseUrl + generatePath(providerUrl, { redirect, providerName});
+
+    if (loginHint) {
+      const url = new URL(basePath);
+      url.searchParams.set("login_hint", loginHint);
+
+      // Remove and add redirect_uri to ensure that it is the last query
```

Comment:
We could even be a little bit more specific here.
```suggestion
      // Remove and add redirect_uri to ensure that it is the last query param.
      // See the comment in lib/web/apiserver.go:ParseSSORequestParams for details.
```